src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/MidEventCallback.java

```diff
@@ -90,6 +90,19 @@ public JsonNode invoke(String caseTypeId,
         return dataJsonNode(content.getData());
     }
 
+    private void removeNextPageFieldData(CaseDetails currentCaseDetails, Integer order, String caseTypeId, String eventId) {
+        if (order != null) {
+            Set<String> wizardPageFields = uiDefinitionRepository
```

Comment:
looking at this now, rather than a `WizardPagesUtils` I think the best thing is the UIDefRepo to return the list of pages wrapped inside a new class `WizardPages` that will then contain all the utility methods to process the pages. But this might be too many changes now. you can ignore it